src/Microsoft.AspNetCore.Sockets/EndPointOptions.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Authorization;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    public class EndPointOptions<TEndPoint> where TEndPoint : EndPoint
+    {
+        public AuthorizationPolicy Policy { get; set; }
```

Comment:
You probably want to take the name, as that would enable scenarios where the policy is resolved dynamically by name which would be blocked if you took the policy instance always.